kclvm/error/src/lib.rs

```diff
@@ -358,27 +358,52 @@ impl ParseError {
 
     /// New a message parse error with span.
     pub fn message(message: String, span: Span) -> Self {
-        ParseError::Message { message, span }
+        ParseError::Message {
+            message,
+            span,
+        }
     }
 }
 
 impl ParseError {
-    /// Convert a parse error into a error diagnostic.
+    /// Convert a parse error into an error diagnostic.
     pub fn into_diag(self, sess: &Session) -> Result<Diagnostic> {
-        let span = match self {
-            ParseError::UnexpectedToken { span, .. } => span,
-            ParseError::Message { span, .. } => span,
-        };
-        let loc = sess.sm.lookup_char_pos(span.lo());
-        let pos: Position = loc.into();
-        Ok(Diagnostic::new_with_code(
-            Level::Error,
-            &self.to_string(),
-            None,
-            (pos.clone(), pos),
-            Some(DiagnosticId::Error(ErrorKind::InvalidSyntax)),
-            None,
-        ))
+        match self {
+            ParseError::UnexpectedToken { span, .. } => {
+                let loc = sess.sm.lookup_char_pos(span.lo());
+                let pos: Position = loc.into();
+                Ok(Diagnostic::new_with_code(
+                    Level::Error,
+                    &self.to_string(),
+                    None,
+                    (pos.clone(), pos),
+                    Some(DiagnosticId::Error(ErrorKind::InvalidSyntax)),
+                    None,
+                ))
+            }
+            ParseError::Message {
+                message,
+                span,
+            } => {
+                let loc = sess.sm.lookup_char_pos(span.lo());
+                let pos: Position = loc.into();
+
+                let mut sb = StyledBuffer::<DiagnosticStyle>::new();
+                let mut errs = vec![];
+                let code_snippet = CodeSnippet::new(span, Arc::clone(&sess.sm));
+                code_snippet.format(&mut sb, &mut errs);
+                let code_line = extract_code_line(&sb);
```

Comment:
Does `(pos.clone(), pos)` contain text information? Why transfer `code_line` here?

Reply:
No `(pos.clone(), pos)` does not contain text information, it just represents the range of the error.

Reply:
What I mean is that there is no need to store both text and location information in the wrong structure. They are redundant in terms of information dimension. We can concentrate on processing the relationship between text and location in the language server, rather than extracting information when constructing the wrong structure.

Reply:
I understand the redundancy here. Let me make the changes. I am in the process of making changes as discussed here https://github.com/kcl-lang/kcl/issues/1125#issuecomment-1999317601 .